tripy/docs/pre0_user_guides/02-compiler.md

```diff
@@ -0,0 +1,191 @@
+# Using the Compiler
+
+```{contents} Table of Contents
+:depth: 3
+```
+
+## Eager Mode vs. Compile Mode
+
+Tripy offers two modes of operation: eager mode and compile mode:
+
+### Eager Mode
+
+Eager mode is designed for development and debugging purposes:
+
+1. **Deferred Compilation**:  In eager mode, compilation is deferred until the first use of a tensor, allowing for easier debugging.
+2. **Fast Compilation**: It utilizes specific settings (`builder_optimization_level`) in the TensorRT compiler to skip most optimizations, resulting in faster compilation times.
+3. **Debugging**: Eager mode is particularly helpful when debugging models that are being developed using Tripy APIs.
+4. **Immediate Feedback**: It allows for quick iterations and immediate feedback, which is beneficial during the model development phase.
+
+### Compile Mode
+
+Compile mode is focused on optimization and deployment:
+
+1. **Full Optimization**: Users can leverage the full power of TensorRT's optimizations to improve model's performance.
+2. **Model Serialization**: Compile mode allows users to serialize the compiled model, enabling deployment and distribution.
+3. **Dynamic Shape Support**: Provides the ability to specify a range of shapes for all inputs, allowing for optimized execution across different input sizes.
```

Comment:
Part of these looks duplicate with "An introduction to Tripy", like sections "Lasy evaluation" and "To compile or not to compile". Maybe this part is better merged with those sections, and `compiler.md` only focuses on how to use `tp.compile`.

Reply:
Removed in the latest iteration.